flatnav/DistanceInterface.h

```diff
@@ -22,8 +22,10 @@ template <typename T> class DistanceInterface {
   // and a query vector. For regular distances (l2, inner product), there is
   // no difference between the two. However, for quantization techniques, such
   // as product quantization, the two distance modes are different.
+
   float distance(const void *x, const void *y, bool asymmetric = false) {
-    return static_cast<T *>(this)->distanceImpl(x, y, asymmetric);
```

Comment:
I'm starting to think we should maybe separate the quantization techniques from the regular distance implementations. It's nice to be able to use CRTP and have a nice interface that can hold quantizers and regular distance objects. One design problem with this interface is that quantizers (PQ, LPQ) need distance implementers (`SquaredL2Distance` and `InnerProductDistance`) as member fields.